### ISSUE TITLE
Keyerror when Running Eva Algorithm

### ISSUE THREAD

**Issue by rudra0713** [GiulioRossetti/cdlib#217]
Hi,
   I am trying to run the algorithm Eva on a small graph. I followed the code snippet mentioned here:
https://cdlib.readthedocs.io/en/latest/reference/cd_algorithms/algs/cdlib.algorithms.eva.html#cdlib.algorithms.eva
```
g = nx.Graph([(1, 2), (1, 3), (2, 5), (3, 4), (3, 6)])

l1 = ['A', 'B', 'C']
for node in g.nodes():
    labels[node] = {"l1": random.choice(l1)}
eva_communities = eva(g, labels, alpha=0.8)
```
However, I am getting the error:
```
status.com_attr[com][label][k] -= status.attr[node][label][k]
KeyError: 'A'
```
The error's origin is at `def __remove()` in `eva.py`. Can anyone kindly explain to me why the algorithm is not working?

**Comment by dsalvaz** [comment]
Hi, I will attempt to resolve the bug.

In the meantime, the algorithm works if you remap the node ids from 0 to n:

```
g = nx.Graph([(1, 2), (1, 3), (2, 5), (3, 4), (3, 6)])  

mapping = dict(zip(g, range(0, len(g))))
relabel_g = nx.relabel_nodes(g, mapping)

l1 = ['A', 'B', 'C']
labels = {}
for node in relabel_g.nodes():
    labels[node] = {"l1": random.choice(l1)}
   
eva_communities = eva(relabel_g, labels, alpha=0.8)
```

**Comment by rudra0713** [comment]
Hi @dsalvaz, Thanks for your comment.
Just wanted to point out, currently, the algorithm is working when the node ids are from 0 to n in that order. For example,
```
g = nx.Graph([(0, 1), (1, 2), (1, 3), (3, 4), (2, 5), (3, 6)])
```
works without relabeling. However, if I define _g_ to be like this (please, note that, the order of edges has been changed)
```
g = nx.Graph([(0, 1), (1, 2), (1, 3), **(2, 5), (3, 4)**, (3, 6)])
```
Then the code fails randomly with the aforementioned error.

But thanks for the temporary fix.

**Comment by GiulioRossetti** [comment]
Fixed in CDlib v0.3.0
